app/src/main/java/com/github/se/wanderpals/ui/screens/docs/DocumentsPS.kt

```diff
@@ -0,0 +1,492 @@
+import com.google.firebase.storage.StorageReference
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+/**
+ * This composable function is used to display the documents of a trip. It displays the documents of
+ * the trip and the documents of the current user.
+ *
+ * @param viewModel the view model of the documents
+ * @param storageReference the reference to the storage
+ */
+const val ERROR_MESS = "You need to fill the name and upload an image"
+
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter", "CoroutineCreationDuringComposition")
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun DocumentsPS(viewModel: DocumentPSViewModel, storageReference: StorageReference?) {
+
+  val context = LocalContext.current
+
+  val documentslistURL by viewModel.documentslistURL.collectAsState()
+  val documentslistUserURL by viewModel.documentslistUserURL.collectAsState()
+
+  // make a topBar to switch between shared and personal documents
+  var state by remember { mutableIntStateOf(0) }
+  val titles = listOf("Private", "Shared")
+
+  var isDisplayed by remember { mutableStateOf(false) }
+  var selectedDocument by remember { mutableStateOf("") }
+  var selectedImagesLocal by remember { mutableStateOf<Uri?>(Uri.EMPTY) }
+  var displayedTheBoxSelector by remember { mutableStateOf(false) }
+  var documentName by remember { mutableStateOf("") }
+  var isUploading by remember { mutableStateOf(false) }
+  var isUploaded by remember { mutableStateOf(false) }
+  var launch by remember { mutableStateOf(false) }
+  var isloading by remember { mutableStateOf(true) }
+
+  var error by remember { mutableStateOf("") }
+
+  val scope = rememberCoroutineScope()
+
+  scope.launch {
+    delay(2000)
+    isloading = false
+  }
+
+  LaunchedEffect(key1 = launch) {
+    if (launch) {
+      isUploading = true
+      delay(3000)
+      isUploaded = true
+      delay(2000)
+      isUploading = false
+      delay(1000)
+      launch = false
+    }
+  }
+
+  // get all the documents from the trip
+
+  LaunchedEffect(Unit) {
+    viewModel.getAllDocumentsFromTrip()
+    viewModel.getAllDocumentsFromCurrentUser()
+  }
+
+  val singlePhotoPickerLauncher =
+      rememberLauncherForActivityResult(
+          contract = ActivityResultContracts.PickVisualMedia(),
+          onResult = { uri ->
+            selectedImagesLocal = uri
+            launch = true
+          })
+
+  Scaffold(
+      modifier = Modifier.fillMaxSize().testTag("documentsScreen"),
+
+      // make a topBar to switch between shared and personal documents
+      // make a list of documents
+      topBar = {
+        SecondaryTabRow(selectedTabIndex = state) {
+          titles.forEachIndexed { index, title ->
+            Tab(
+                modifier = Modifier.height(70.dp).testTag("tab$title"),
+                text = { Text(title) },
+                selected = state == index,
+                onClick = { state = index })
```

Comment:
Maybe modify the top bar to look like the rest of the app with a blue background and white text (primary and onPrimary)

Reply:
Well done!